### PR TITLE
Contribution guidelines and pull request template addition

### PR DESCRIPTION
**Pull request by Vedant1202**
Added a draft for contribution guidelines and pull request template.

### CODE REVIEW

**Comment by cjchirag7** [comment]
@Vedant1202 , this looks nice ! We really needed a contributing.md  
but you may wish to change the base of your PR to `develop` branch, as mentioned in the developer guide ( See README.md )

**Comment by Vedant1202** [comment]
@cjchirag7 Thanks for the review. There is no change in the system as such it's just an addition as a documentation context. If it's reviewed and passed by the maintainers, I think merging into `master` should'nt be a problem. 
Also, since contributions are forked and branched from the default branch, ie, `master`, and github even displays the default branch when someone visits the repo, I guess adding it directly to the default branch would be much more better.

**Comment by Vedant1202** [comment]
@cjchirag7 I have however, missed adding the line where you've to request to pull a merge into the `develop` branch in the contrib file. I'll do that.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+- Proposing new features
+- Becoming a maintainer
+
+## We Develop with Github
+We use github to host code, to track issues and feature requests, as well as accept pull requests.
+
+## We Use [Github Flow](https://guides.github.com/introduction/flow/index.html), So All Code Changes Happen Through Pull Requests
+Pull requests are the best way to propose changes to the codebase (we use [Github Flow](https://guides.github.com/introduction/flow/index.html)). We actively welcome your pull requests:
+
+1. Fork the repo and create your branch from `master`.
```
Comment (cjchirag7) [inline CONTRIBUTING.md:16]:
```suggestion
1. Fork the repo and create your branch from `develop`.
```
This should be develop, as per the current contributing guidelines.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+2. If you've added code that should be tested, add tests.
+3. If you've changed APIs, update the documentation.
+4. Ensure the test suite passes.
+5. Make sure your code lints.
```
Comment (cjchirag7) [inline CONTRIBUTING.md:20]:
```suggestion
5. Make sure your code passes the lint tests, by running `npm run lint-fix` ( after installing the dependencies using `npm install` ). 
```

Comment (Vedant1202) [reply CONTRIBUTING.md:20]: @cjchirag7, I think so this has been mentioned in a seperate [section](https://github.com/Vedant1202/caMicroscope/blob/contrib-PR/CONTRIBUTING.md#use-a-consistent-coding-style). I'll reference to that section though

**Comment by Vedant1202** [comment]
@cjchirag7 I've made the changes, thanks for the suggestions.

**Comment by birm** [comment]
Hi, thanks for putting this together.
I'd recommend the PR template to go to https://github.com/camicroscope/.github so it applies to all caMicroscope repos.
The biggest thing that I think needs discussion/consideration is the workflow we're using. I think we're somewhere between the link you posted and gitflow, since we removed the release branch but kept the develop. I think there's merit to removing develop and just use tags. But I don't want to make that decision unilaterally.

**Comment by Vedant1202** [comment]
@birm I have shifted PR template to .github. 
I saw a discussion is opened on [PR-366](https://github.com/camicroscope/caMicroscope/pull/366), so I guess we'll have to wait until we get to a conclusion over there?

**Comment by birm** [comment]
I think it's ok to discuss here or in the issue. I just don't want to forget to actually do a release this week, before there are too many changes. (Also, as you pointed out, the contributing guidelines don't go into effect until we merge)

**Comment by birm** [comment]
After talking with other contributors, It looks like we want to keep develop, so that checking out master is the same as checking out the latest release.

**Pull request merged**